.github/shared/test/spec-model.test.js

```diff
@@ -0,0 +1,349 @@
+      });
+
+      const actual = getAffectedSwaggers(
+        "specification/1/data-plane/d.json",
+        specModel,
+      );
+      const expected = new Set([
+        "specification/1/data-plane/a.json",
+        "specification/1/data-plane/nesting/b.json",
+        "specification/1/data-plane/c.json",
+        "specification/1/data-plane/d.json",
+      ]);
+      expect(actual).toEqual(expected);
+    },
+  );
+
+  it.skipIf(isWindows)(
+    "returns correctly for multiple shared dependencies",
+    async ({ expect }) => {
+      const specModel = await getSpecModel("specification/1", {
+        repoRoot: resolve(__dirname, "fixtures/getAffectedSwaggers"),
+      });
+
+      const actual = getAffectedSwaggers(
+        "specification/1/data-plane/shared/shared.json",
+        specModel,
+      );
+      const expected = new Set([
+        "specification/1/data-plane/a.json",
+        "specification/1/data-plane/nesting/b.json",
+        "specification/1/data-plane/c.json",
+        "specification/1/data-plane/d.json",
+        "specification/1/data-plane/shared/shared.json",
+        "specification/1/data-plane/e.json",
+      ]);
+      expect(actual).toEqual(expected);
+    },
+  );
+});
+
+// Stress test the parser against all specs in the specification/ folder. This
+// is a long-running test and should be run manually. To run this test, remove
+// the '.skip' from the describe block. Put '.skip' back in when done or this
+// test may fail unexpectedly in the future.
+describe.skip("Parse readmes", () => {
+  it(
+    "Does not produce exceptions",
+    { timeout: 30 * 60 * 1000 /* 30 minutes */ },
+    async ({ expect }) => {
+      const excludeFolders = [
+        "authorization", // specification/authorization/resource-manager/readme.md defines has duplicate tags including 'package-2020-10-01'
+        "azureactivedirectory", // specification/azureactivedirectory/resource-manager/readme.md has duplicate tags including 'package-preview-2020-07'
+        "cost-management", // specification/cost-management/resource-manager/readme.md has duplicate tags including 'package-2019-01'
+        "migrate", // specification/migrate/resource-manager/readme.md has duplicate tags including 'package-migrate-2023-04'
+        "quota", // specification/quota/resource-manager/readme.md has duplicate tags including 'package-2023-02-01'
+        "redisenterprise", // specification/redisenterprise/resource-manager/readme.md has duplicate tags including 'package-2024-02'
+        "security", // specification/security/resource-manager/readme.md has duplicate tags including 'package-2021-07-preview-only'
+        "confidentialledger", // data-plane/readme.md tag 'package-2022-04-20-preview-ledger' points to a swagger file that doesn't exist
+        "network", // network takes a long time to evaluate
+        "servicenetworking", // servicenetworking includes a swagger file which references a file that doesn't exist
+      ];
```

Comment:
Many of these services have readmes with multiple definitions of tags which, after consulting with @timotheeguerin, is an error. At this point the spec model throws an error. 

It might make sense to add some kind of `useLastDefinitionOfTagInputFiles` option that enables the parser to keep going forward. Testing on Legacy LintDiff shows this kind of behavior.